docs/tools/confirmation.md

```diff
@@ -0,0 +1,222 @@
+    [known limitations](#known-limitations).
+    We welcome your
+    [feedback](https://github.com/google/adk-python/issues/new?template=feature_request.md&labels=tool%20confirmation)!
+
+You can configure how a request is communicated to a user, and the system can
+also use [remote responses](#remote-response) sent via the ADK
+server's REST API. When using the confirmation feature with the ADK web user
+interface, the agent workflow displays a dialog box to the user to request
+input, as shown in Figure 1:
+
+![Screenshot of default user interface for tool confirmation](/adk-docs/assets/confirmation-ui.png)
+
+**Figure 1.** Example confirmation response request dialog box using an
+advanced, tool response implementation.
+
+The following sections describe how to use this feature for the confirmation
+scenarios. For a complete code sample, see the
+[human_tool_confirmation](https://github.com/google/adk-python/blob/fc90ce968f114f84b14829f8117797a4c256d710/contributing/samples/human_tool_confirmation/agent.py)
+example. There are additional ways to incorporate human input into your agent
+workflow, for more details, see the
+[Human-in-the-loop](/adk-docs/agents/multi-agents/#human-in-the-loop-pattern)
+agent pattern.
+
+## Boolean confirmation {#boolean-confirmation}
+
+When your tool only requires a simple `yes` or `no` from the user, you can
+append a confirmation step using the `FunctionTool` class as a wrapper. For
+example, if you have a tool called `reimburse`, you can enable a confirmation
+step by wrapping it with the `FunctionTool` class and setting the
+`require_confirmation` parameter to `True`, as shown in the following example:
+
+```
+# From agent.py
+root_agent = Agent(
+   ...
+   tools=[
+        # Set require_confirmation to True to require user confirmation
+        # for the tool call.
+        FunctionTool(reimburse, require_confirmation=True),
+    ],
+...
+```
+
+This implementation method requires minimal code, but is limited to simple
+approvals from the user or confirming system. For a complete example of this
+approach, see the
+[human_tool_confirmation](https://github.com/google/adk-python/blob/fc90ce968f114f84b14829f8117797a4c256d710/contributing/samples/human_tool_confirmation/agent.py)
+code sample.
+
+### Require confirmation function
+
+You can modify the behavior `require_confirmation` response by replacing its
+input value with a function that returns a boolean response. The following
+example shows a function for determining if a confirmation is required:
+
+```
+async def confirmation_threshold(
+    amount: int, tool_context: ToolContext
+) -> bool:
+  """Returns true if the amount is greater than 1000."""
+  return amount > 1000
+```
+
+This function than then be set as the parameter value for the
+`require_confirmation` parameter:
+
+```
+root_agent = Agent(
+   ...
+   tools=[
+        # Set require_confirmation to True to require user confirmation
```

Comment:
Update the comment here?  Something like: 

"Require confirmation when the amount exceed the confirmation threshold"